build_xlsx.py

```diff
@@ -60,6 +61,124 @@ def write_data(workbook, worksheet, buckets):
     })
 
 
+def write_header_policies(workbook, worksheet):
+    header_format = workbook.add_format()
+    header_format.set_bold()
+    header_format.set_align("center")
+    header_format.set_align("vcenter")
+    header_format.set_border()
+
+    worksheet.set_column("A:A", 45)
+    worksheet.set_column("B:B", 25)
+    worksheet.set_column("C:C", 10)
+    worksheet.set_column("D:D", 45)
+    worksheet.set_column("E:E", 20)
+    worksheet.set_column("F:F", 100)
+    worksheet.merge_range("E1:F1", "Principal", header_format)
+    worksheet.write("B1", "Actions", header_format)
+    worksheet.write("C1", "Effect", header_format)
+    worksheet.write("D1", "Resources", header_format)
+    worksheet.write("A1", "S3 Buckets", header_format)
+
+
+def format_sheet(worksheet, workbook, line):
+    red = workbook.add_format()
+    red.set_bg_color("#ff948a")
+    red.set_bold()
+    green = workbook.add_format()
+    green.set_bg_color("#b7e1cd")
+    worksheet.conditional_format(1, 1, line, 11, {
+        "type":     "text",
+        "criteria": "containing",
+        "value":    "Deny",
+        "format":   green,
+    })
+    worksheet.conditional_format(1, 1, line, 11, {
+        "type":     "text",
+        "criteria": "containing",
+        "value":    "Allow",
+        "format":   red,
+    })
+
+def write_right_principal(worksheet, policies, data_format, i, color_format):
+    try:
```

Comment:
You should avoid chaining try / catch like this

Reply:
fixed